File: java/src/main/java/com/languageweaver/sdk/samples/linguisticOptions/EdgeLinguisticOptionsService.java
package com.languageweaver.sdk.samples.linguisticOptions;

import com.languageweaver.sdk.common.edge.EdgeLanguageWeaverClient;
import com.languageweaver.sdk.translate.edge.request.linguisticOptions.EdgeGetLinguisticOptionsRequest;
import com.languageweaver.sdk.translate.edge.result.linguisticOptions.EdgeLinguisticOptionsResult;

public class EdgeLinguisticOptionsService {

    public static void main(String[] args) throws Exception {
        try (EdgeLanguageWeaverClient lwClient = new EdgeLanguageWeaverClient().build()) {
            EdgeGetLinguisticOptionsRequest edgeGetLinguisticOptionsRequest = new EdgeGetLinguisticOptionsRequest()
                    .setSourceLanguageId("chi")
                    .setTargetLanguageId("eng")
                    .setModel("Generic");
//                   .setPlatform("SRV")
//                   .setTechnology("TNMV")

            final EdgeLinguisticOptionsResult edgeLinguisticOptionsResult = lwClient.getEdgeLinguisticOptions(edgeGetLinguisticOptionsRequest);
            // handle result
        }
    }

}

File: javascript/browser/src/samples/linguisticOptions/EdgeLinguisticOptionsService.js
import {
    ClientConfiguration,
    CredentialsConfiguration,
    EdgeGetLinguisticOptionsRequest,
    EdgeLanguageWeaverClient
} from "@language-weaver/lw-sdk-js";

export const getEdgeLinguisticOptions = async clientId => {
    try {
        const clientConfiguration = new ClientConfiguration();
        clientConfiguration.credentialsConfiguration = new CredentialsConfiguration(clientId);
        const edgeLanguageWeaverClient = await new EdgeLanguageWeaverClient()
            .withConfigurations(clientConfiguration)
            .build();
        const request = new EdgeGetLinguisticOptionsRequest();
        request.sourceLanguageId = "chi";
        request.targetLanguageId = "eng";
        request.model = "Generic";
        // request.platform = "SRV";
        // request.technology = "TNMV";
        const linguisticOptionsResult = await edgeLanguageWeaverClient.getEdgeLinguisticOptions(request);
        console.log(linguisticOptionsResult.linguisticOptions);
    } catch (e) {
        console.log(e);
    }
}